### DISCUSSION TITLE
Text Field/Editor Key Mapping Shift + Home

### DISCUSSION THREAD

**Discussion by heejit** [cogentcore/core#1282] [General]
is there universal key mapping exists ?

I want to select current Line from current post to start of line using shift + home key do I have to use KeyChord event ??

**Comment by kkoreilly** [comment]
That is a good idea; we will add support for that to the base text field and text editor soon.

**Reply by heejit** [reply]
Shift + Home key select all text from current cursor pos to the start not the start-of-line is this a normal behavior ?

**Reply by kkoreilly** [reply]
I tested it and Shift + Home selecting all the way to the full start seems to be standard behavior, at least in GitHub and gmail. Therefore, we just need to add support for Shift + Home and Shift + End in TextField and then our behavior will be standard.

**Reply by heejit** [reply]
I am using firefox and Shift +Home in this comment box selecting start of line.

and any desktop app on my laptop(opensuse) also same.

**Reply by heejit** [reply]
Is there any shortcut in TextEditor to select start and end of line from current position ??

**Reply by kkoreilly** [reply]
Ctrl+Shift+Left and Ctrl+Shift+Right should work for that purpose, but it looks like they are not currently working, so I will add support for those.

**Reply by heejit** [reply]
Is there any exposed function of texteditor which I can use it  to detect cursor pos and select  ??

**Reply by heejit** [reply]
This shortcut I am using for all other text editing app and many time mistakenly I deleted whole text because it work differently in core.texteditor.

**Reply by heejit** [reply]
Maybe a api like this ->
`texteditor.SetShortcut(ActionName, Shortcutkey)`

**Reply by kkoreilly** [reply]
Thank you for the feedback and apologies for the issue. We will add a setting or API such that you have the desired behavior.

**Reply by kkoreilly** [reply]
Okay, I did some more investigation about how this works on different platforms, and I was incorrect about the right behavior; you are correct. This is what we will implement, which is consistent with what you suggested:

* `Home/End` -> go to start/end of line
* `Shift`+`Home/End` -> select to start/end of line
* `Ctrl/Cmd` + `Shift` + `Left/Right` -> select to start/end of line
* `Ctrl/Cmd/Alt` + `Home/End` -> go to start/end of document

Thank you again for your feedback and apologies for the incorrect keybindings. We will implement the correct keybindings now.

**Reply by heejit** [reply]
Thanks

**Reply by kkoreilly** [reply]
As of #1300, all of these keybindings should be working correctly. Apologies for the delay in implementing this.

**Reply by heejit** [reply]
How do I update ?

**Reply by rcoreilly** [reply]
You can put the latest hash: 06fac9d9efbcf4ef8f7c772b77483f821d53db95 in your go.mod file for cogentcore.org/core and do go mod tidy.

**Reply by heejit** [reply]
like this -> cogentcore.org/core v0.3.5-06fac9d

**Reply by heejit** [reply]
this work -> go get cogentcore.org/core@06fac9d

**Reply by heejit** [reply]
key working as expected thanks for quick update.
